src/model/PameldingFormValues.ts

```diff
@@ -2,19 +2,67 @@ import {z} from 'zod'
 import {DeltakelsesprosentValg, MAL_TYPE_ANNET} from '../utils.ts'
 import {Mal, PameldingResponse} from '../api/data/pamelding.ts'
 
-export const pameldingFormSchema = z.object({
-  valgteMal: z.array(z.string()),
-  malAnnetBeskrivelse: z.string().optional(),
-  bakgrunnsinformasjon: z.string()
-    .max(500, 'Kan ikke være mer enn 500 tegn'),
-  deltakelsesprosentValg: z.nativeEnum(DeltakelsesprosentValg).optional(),
-  deltakelsesprosent: z.number().optional(),
-  dagerPerUke: z.number().optional()
-})
+export const BESKRIVELSE_MAX_TEGN = 250
+export const BAKGRUNNSINFORMASJON_MAKS_TEGN = 1000
+
+export const pameldingFormSchema = z
+  .object({
+    valgteMal: z.string().array().nonempty('Du må velge innhold'),
+    malAnnetBeskrivelse: z
+      .string()
+      .max(
+        BESKRIVELSE_MAX_TEGN,
+        `Beskrivelse for mål Annet kan ikke være mer enn ${BESKRIVELSE_MAX_TEGN} tegn`
+      ),
+    bakgrunnsinformasjon: z
+      .string()
+      .min(1, { message: 'Du må skrive noe for mål Annet' })
+      .max(
+        1000,
+        `Bakgrunnsinformasjon kan ikke være mer enn ${BAKGRUNNSINFORMASJON_MAKS_TEGN} tegn`
+      ),
+    deltakelsesprosentValg: z.nativeEnum(DeltakelsesprosentValg).optional(),
+    deltakelsesprosent: z
+      .number({
+        invalid_type_error: 'Deltakelsesprosent må være ett tall mellom 0 og 100.'
+      })
+      .int({ message: 'Deltakelsesprosent må være et heltall' })
+      .gte(0, { message: 'Deltakelsesprosent må være større enn 0' })
+      .lte(100, { message: 'Deltakelsesprosent må være mindre enn 100' })
+      .optional(),
+    dagerPerUke: z
+      .number()
+      .int({ message: 'Dager per uke må være et heltall' })
+      .gte(0, { message: 'Dager per uke må være større enn 0' })
```

Comment:
Den minste verdien for `dagerPerUke` og `deltakelsesprosent` bør nok også være 1. Det er litt rart å delta på et tiltak 0 dager i uken 😀

Reply:
Enig :)